cdap-ui/app/cdap/components/BtnWithLoading/index.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import LoadingSVG from 'components/LoadingSVG';
+require('./BtnWithLoading.scss');
+
+export default function BtnWithLoading({
+  className,
+  label,
+  loading,
+  disabled,
+  onClick
+}) {
+  return (
+    <button
+      className={`btn btn-with-loading ${className}`}
+      onClick={onClick}
+      disabled={disabled || loading}
+    >
+      {
+        !loading ? label : (
+          <div>
+            <LoadingSVG />
+            {label}
```

Comment:
So far we use `LoadingSVG` to show that a whole page is loading, while for buttons we use `<IconSVG name="icon-spinner" className="fa-spin" />`. Do we want to use `LoadingSVG` even on buttons from now on? Also for buttons we usually have the loading icon after (to the right of) the label.

Reply:
Yes. Since we have adopted to use font icons we should use `LoadingSVG`. I noticed this pattern in `DataPrepServiceControl` component and it had the loading icon to the left and not to the right. Hence the ordering.